### PR TITLE
ADBDEV-2818: Provide upgrade scripts for diskquota extension

### PR DESCRIPTION
**Pull request by RekGRpth**
ADBDEV-2818: Provide upgrade scripts for diskquota extension

Merge tag '2.0.0' into ADBDEV-2818

### CODE REVIEW

**Review by Stolb27 — CHANGES_REQUESTED** [review]
@RekGRpth, do we really have to modify existing 1.0 version? If there are no strong reasons, we should leave it unchanged.
diskquota installed from original 1.0 version should do the same as downgraded from 2.0.
We should do that to get predictable behavior for both cases: the cluster was updated from original 1.0 or 1.0.3 and the cluster was installed with 2.0 from scratch. With the current implementation, you get a different set of sql entities.

Also, please provide appropriate commit message and PR description.

**Comment by Stolb27** [comment]
> @RekGRpth, do we really have to modify existing 1.0 version? If there are no strong reasons, we should leave it unchanged.
diskquota installed from original 1.0 version should do the same as downgraded from 2.0.
We should do that to get predictable behavior for both cases: the cluster was updated from original 1.0 or 1.0.3 and the cluster was installed with 2.0 from scratch. With the current implementation, you get a different set of sql entities.

As we discussed previously, we decided to sync 1.0 file content to avoid conflicts. This version is unusable anyway as far as library for 2.0 is incompatible, and we aren't going to vendor 1.0 library with 2.0 together.

Update from 1.0 and from 1.0.3 is identical because the upstream has reverted sql function `diskquota.update_diskquota_db_list` creation and remove all its sql usages. All changes between our 1.0 and 1.0.3 diskquota--1.0--1.0.3.sql.

File: diskquota--1.0.3--2.0.sql [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+);
+-- TODO ALTER TABLE diskquota.target SET DEPENDS ON EXTENSION diskquota;
+
+ALTER TABLE diskquota.table_size ADD COLUMN segid smallint DEFAULT -1; -- segid = coordinator means table size in cluster level
+ALTER TABLE diskquota.table_size DROP CONSTRAINT table_size_pkey;
+ALTER TABLE diskquota.table_size ADD PRIMARY KEY (tableid, segid);
+ALTER TABLE diskquota.table_size SET WITH (REORGANIZE=true) DISTRIBUTED BY (tableid, segid);
+
+-- TODO SELECT pg_catalog.pg_extension_config_dump('diskquota.target', '');
+-- TODO SELECT gp_segment_id, pg_catalog.pg_extension_config_dump('diskquota.target', '') FROM gp_dist_random('gp_id');
```
Comment (Stolb27) [inline diskquota--1.0.3--2.0.sql]: @RekGRpth, can you explain why they can't mark newly created table as needed to dump? This table will have different attributes for clean 2.0 install and for 2.0 after upgrade.

Comment (RekGRpth) [reply diskquota--1.0.3--2.0.sql]: I have no any idea why updating does not mark new table `diskquota.target` as needed to dump, while clean installing does it.

Comment (Stolb27) [reply diskquota--1.0.3--2.0.sql]: the comment from lines was removed

**Review by Stolb27 — APPROVED** [review]

**Comment by Stolb27** [comment]
@deart2k, please take a look at diskquota-2.0 sync. If there is no problem, you can merge and tag it.

**Review by deart2k — APPROVED** [review]

**Pull request merged**